libutils/json.h

```diff
@@ -160,6 +161,37 @@ JsonElement *JsonCopy(const JsonElement *json);
 int JsonCompare(const JsonElement *a, const JsonElement *b);
 JsonElement *JsonMerge(const JsonElement *a, const JsonElement *b);
 
+/**
+ * @brief Recusivly merge one object into another
+ *
+ * @param base Object to merge data into
+ * @param extra Object to merge data from
+ * @return Pointer to base object (convenient for nested function calls)
+ *
+ * @note Function should not return NULL
+ * @warning Side effects: Base object is modified in-place. If this is not the
+ *          intension, consider using JsonObjectMergeDeep
+ */
+JsonElement *JsonObjectMergeDeepInplace(JsonElement *base, const JsonElement *extra);
+
+/**
+ * @brief Recusivly merge two objects into a new object
+ *
+ * @param base Object to copy and merge data into
+ * @param extra Object to merge data from
+ * @return Pointer to merged object
+ *
+ * @note Function should not return NULL
+ * @warning Performance penalty: A new object is created. If this is not the
+ *          intension, consider using JsonObjectMergeDeepInplace
+ */
+static inline JsonElement *JsonObjectMergeDeep(const JsonElement *const base, const JsonElement *const extra)
+{
+    assert(base != NULL);
+    JsonElement *const copy = JsonCopy(base);
+    return JsonObjectMergeDeepInplace(copy, extra);
```

Comment:
This could do a little better job by constructing the `JsonElement` while merging `base` and `extra`. Not sure if it's worth the extra complexity, though. Maybe a comment would do for now.

Reply:
Indeed it could be better, although I like the simplicity. Added a comment for now :)